### DISCUSSION TITLE
Any testers out there? [Custom Quality Definitions for x264/x265]

### DISCUSSION THREAD

**Discussion by Dendraspis** [staxrip/staxrip#451] [Ideas]
Hey  @44vince44  and  @JJKylee ,
hello everybody,

are you interested in testing a new feature?  😉 
It's one nobody needs except me.  🤣 

<img src="https://user-images.githubusercontent.com/50659978/104766705-4ecb5e00-576b-11eb-885c-ec17939e1028.png">
<img src="https://user-images.githubusercontent.com/50659978/104766706-4f63f480-576b-11eb-9cf2-c3c03d6a6f85.png">

Known issue:  
- After changing the settings you have to change the video profile and change back to see the new values.
- Because of limited space and to keep it simple the definition blocks are kept very short, but it should be okay for the experts who wants/needs to use this feature.

[StaxRip v2.1.7.1_DDef.zip](https://github.com/staxrip/staxrip/files/5822460/StaxRip.v2.1.7.1_DDef.zip)

**Comment by stax76** [comment]
It's a good feature, but I think the definition is not very intuitive, suggesting something like this:

name = value; name2 = value2

**Reply by Dendraspis** [reply]
I know what you mean, but because of the small size you loose the overview very quickly. Also I wanted to make it possible to edit those values very quickly and without the need to enter a text/description. So my solution is/was using a number followed by an optional text surrounded by quotation marks.

This might be difficult for the first run and maybe you have to read the help text (which is quite long :grin:), but when you know what you do, this is extremely fast.

For example this is possible:
`11 12 15 16 18 19 20 21`
...or...
`8"Crazy" 10"Why not?!" 12 13,5 15.5 18.7"why?"`
...or...
`8"Crazy"__11   13   15   17  18  20`

An example can be find in the help text, too. :wink:

Nevertheless this is something you are going to set once and maybe adjust twice - that's it. So I don't think we have to blow it up, especially when I'm the only one using it. :grin:

**Comment by JJKylee** [comment]
Well, more options are good, and friendly looking definitions may be a plus for users.

That said, **I** might _not_ use it at all since there's already a good set of pre-selected `crf` values for _quality_ in the video encoder pane.

![StaxRip_2 1 7 1_DDef_without_quality-definitions_20210115](https://user-images.githubusercontent.com/3157518/104768945-5535e780-5723-11eb-9490-87a1ef091220.jpg)

And making my own quality definitions in `Settings` > `Video` is kind of cumbersome. You need to simplify the process/syntax. 😋

But who knows? Even if I wouldn't use it, there may be others who might like it a lot.

Good try. 👏

**Reply by Dendraspis** [reply]
> You need to simplify the process/syntax. 😋

:thinking: You mean something like configuration via Alexa?  :grin:

But like you already mentioned, it might be an option for nerds. 🤓 So no need to change something. 😋

**Reply by JJKylee** [reply]
Got it.

Then how about changing the option names like
x265 quality definitions ⇨ x265 quality _**custom**_ definitions?
Would it occupy too much space? 😓

**Reply by Dendraspis** [reply]
> Would it occupy too much space? 😓

I think yes. I want to look after the textbox and why it doesn't expand any further, because there is a gap.  🤔  But that's something we can adjust later.  😄

**Comment by stax76** [comment]
There is a non ANSI char in the changelog it seems:

- Introduce custom quality definitions for x264 and x265 via Settings � Video (Dendraspis)

For this to work I think you must first tell VS to use Unicode for this file, but I think we don't need non ASCII and because of that there is a check in the release script that verifies the complete code is ASCII:

https://github.com/staxrip/staxrip/blob/master/Release.ps1#L38

But this is not important for me, change it and use Unicode if you want to.

**Reply by Dendraspis** [reply]
Worked for me when I typed it in, but after merging I see the same symbol.

No need to, I could use a variety of other chars.  😁  Is already fixed.

**Comment by stax76** [comment]
I've improved the layout.

**Reply by Dendraspis** [reply]
Seems like it was a self-induced copy&paste problem.  😁  🤣 

Thanks for that! Wanted to look after this in a few hours.  😄 

---------------
@stax76 
Not the right place, but could you also update [NVEnc 5.25](https://github.com/rigaya/NVEnc/releases/tag/5.25) for the next beta, please? Currently we still use v5.22.

**Comment by stax76** [comment]
@Dendraspis 

I've updated nvenc with the auto update and have subscribed to release notifications.

**Comment by ghost** [comment]
Sorry, I'm still not used to check in the Discussion, especially because there are no "unread" markers...
But I'm definitely trying to remember to check there.

**I think this feature is very important**, because I share @Dendraspis 's idea that the current labels are not totally adequate.
For instance, as per my requirements, crf 24 should be "medium"  instead of "very low"...

Is there any thing updated to test ? Or am I too late ? (again, I'm terribly sorry, I didn't check here until now)

BTW I'm getting no notifications. is it because I'm using the browser on my laptop and not the app ?

**Reply by JJKylee** [reply]
> **I think this feature is very important**, because I share @Dendraspis 's idea that the current labels are not totally adequate.
> For instance, as per my requirements, crf 24 should be "medium" instead of "very low"...

You've got a point. Yes, sometimes I also feel the labels have too high a standard, although I ignore the labels themselves. 😆
 
> BTW I'm getting no notifications. is it because I'm using the browser on my laptop and not the app ?

![GitHub_StaxRip_Discussions_notification_switch_20210116](https://user-images.githubusercontent.com/3157518/104820861-ac53bf00-57ec-11eb-805f-bb17230ab8f9.jpg)

You can customize the notifications on `Watch`.

**Reply by Dendraspis** [reply]
> Sorry, I'm still not used to check in the Discussion, especially because there are no "unread" markers...
> But I'm definitely trying to remember to check there.

Therefore I have @mentioned you.  😉 

> **I think this feature is very important**, because I share @Dendraspis 's idea that the current labels are not totally adequate.
> For instance, as per my requirements, crf 24 should be "medium" instead of "very low"...

🤝 

> Is there any thing updated to test ? Or am I too late ? (again, I'm terribly sorry, I didn't check here until now)

The ZIP in the first post is still up to date.  Meanwhile the feature is also commited, so it will be available since v2.1.7.2, but it would be nice if someone could test it (except me) so I could fix issues before the release.

> BTW I'm getting no notifications. is it because I'm using the browser on my laptop and not the app ?

You should get @mentions for discussions, too.
Via Android app it is possible now since the last update and via browser it was always possible to see @mentions and new discussions (if selected) via notifications.  🤔

**Reply by JJKylee** [reply]
> The ZIP in the first post is still up to date. Meanwhile the feature is also commited, so it will be available since v2.1.7.2, but it would be nice if someone could test it (except me) so I could fix issues before the release.

@stax76 has also put `StaxRip.exe` on [his Dropbox archive](https://www.dropbox.com/sh/4ctl2y928xkak4f/AAADEZj_hFpGQaNOdd3yqcAHa?dl=0&lst=). It also has quality definition settings. Now whose is more recent? 👀

**Reply by ghost** [reply]
I'll test this now! (the exe provided by @Dendraspis )

@JJKylee thanks for showing me the custom settings (it was there all the time, and I never saw it!)

**Comment by ghost** [comment]
@Dendraspis this seems to work nicely ;-)))) 

I have nothing to say, it's very good 😄 👍 
(I did a couple of quick encodes just to check it's OK)

**Reply by Dendraspis** [reply]
Thanks for testing.

Seems like I'm not the only person using it.  😁

**Comment by ghost** [comment]
![SNAG-0000](https://user-images.githubusercontent.com/17828567/104827592-e910cc00-5867-11eb-9b7b-bb51be6f78a3.jpg)

My vision 😉 

And thanks to you @Dendraspis !!!
